tools/cache_creator/CMakeLists.txt

```diff
@@ -0,0 +1,53 @@
+##
+ #######################################################################################################################
+ #
+ #  Copyright (c) 2020 Google LLC. All Rights Reserved.
+ #
+ #  Permission is hereby granted, free of charge, to any person obtaining a copy
+ #  of this software and associated documentation files (the "Software"), to deal
+ #  in the Software without restriction, including without limitation the rights
+ #  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ #  copies of the Software, and to permit persons to whom the Software is
+ #  furnished to do so, subject to the following conditions:
+ #
+ #  The above copyright notice and this permission notice shall be included in all
+ #  copies or substantial portions of the Software.
+ #
+ #  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ #  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ #  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ #  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ #  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ #  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ #  SOFTWARE.
+ #
+ #######################################################################################################################
+add_library(cache_creator_lib INTERFACE)
+
+list(APPEND CMAKE_MODULE_PATH "${XGL_LLVM_BUILD_PATH}/lib/cmake/llvm")
+include(LLVMConfig)
```

Comment:
Does this LLVM module have an include_guard?

Reply:
How can I check this? Can you point me to some relevant documentation?

Reply:
I'm not sure I understand what your concern is here.

Reply:
https://cmake.org/cmake/help/latest/command/include_guard.html?highlight=include_guard

I added a link to include_guard essentially it's the cmake equivalent of pragma once.

I'm just worried about inflating configuration times.

Reply:
You'd have to check the cmake code for LLVMConfig.cmake

Reply:
Having short configuration times is crucial for development workflows, and it really shouldn't be treated as an afterthought. Build system code is code is just like any other code that needs to be maintained. And ideally fast.

Reply:
I looked at `LLVMConfig.cmake` and it doesn't use `include_guard`. It looks pretty short to me though, and only includes one other cmake file (LLVM-Config.cmake), which is also concise.
I think this file will be processed twice: once by LLPC, and one by cache-creator. Since LLPC is not in the parent scope of cache-creator, I think there's no way avoid re-including it?

Reply:
Don't worry about it for now. It's out of scope for this review. 
I've always been a bit worried about the configure time for LLVM, for projects that consume LLVM.

However, since you work at Google on the LLVM team perhaps you could take some time in another review to reduce the configure time for LLVM?

Cmake now has profiling capabilities to help out with issues like this:
https://cmake.org/cmake/help/latest/release/3.18.html?highlight=profiling#command-line

I've found it very beneficial for configure times to use include_guard where appropriate. So I just wanted to point it out.